Assets/Scripts/StarComponent.cs

```diff
@@ -17,7 +17,10 @@ public class StarComponent : MonoBehaviour
     void Start()
     {
         dataControllerObj = GameObject.Find("DataController");
-        dataController = dataControllerObj.GetComponent<DataController>();
+        if (dataControllerObj)
+        {
+            dataController = dataControllerObj.GetComponent<DataController>();

```

Comment:
I didn't think truthy-if worked in C#? Regardless, why not simplify this:
`
if (dataController == null) dataController = FindObjectOfType<DataController>();
`

Reply:
Yeah, that works.  You can just test for the object without a comparison to null.

Reply:
good to know! In which case, let's use:
`if (!dataController) dataController = FindObjectOfType<DataController>();`

Reply:
But basically I'm saying only if we can find the game object named DataController, do we try to get a ref to the script on the object.  So I'm talking about both the object and the script on the object.